### PR TITLE
[Non-Forked] Source Pipedrive : Migrate Python CDK to Low-code CDK

### PR DESCRIPTION
**Pull request by archangelic**
## What
Non-forked branch for PR #29293 

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: docs/integrations/sources/pipedrive-migrations.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Pipedrive Migration Guide
+
+## Upgrading to 2.0.0
+Please update your config. This version has changed the config to only require an API key.
```
Comment (marcosmarxm) [inline docs/integrations/sources/pipedrive-migrations.md:4]: update your config and reset your data (to match the new format)

File: airbyte-integrations/connectors/source-pipedrive/source_pipedrive/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    - "deals"
+
+spec:
+  type: Spec
+  documentation_url: https://docs.airbyte.com/integrations/sources/pipedrive
+  connection_specification:
+    title: Pipedrive Spec
+    type: object
+    required:
+      - replication_start_date
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-pipedrive/source_pipedrive/manifest.yaml:343]: `api_token` should be required.

File: airbyte-integrations/connectors/source-pipedrive/source_pipedrive/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+    primary_key: "id"
+    retriever:
+      $ref: "#/definitions/retriever_increment"
+    incremental_sync:
+      $ref: "#/definitions/incremental_base"
+    $parameters:
+      path_extractor: "product"
+      path: "/recents"
+
+  products_fields_stream:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-pipedrive/source_pipedrive/manifest.yaml:245]: This must be `product_fields`  not `products_fields`.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments.

**Comment by airbyte-oss-build-runner** [comment]
##  source-pipedrive test report (commit [`d97b030851`](https://github.com/airbytehq/airbyte/pull/30553/commits/d97b030851f257554e82599134b10efebec8b571)) - ✅

⏲️  Total pipeline duration: 02mn03s 

| Step                                                                    | Result   |
|:------------------------------------------------------------------------|:---------|
| Connector package install                                               | ✅        |
| Build source-pipedrive docker image for platform linux/x86_64           | ✅        |
| Acceptance tests                                                        | ✅        |
| Code format checks                                                      | ✅        |
| Validate airbyte-integrations/connectors/source-pipedrive/metadata.yaml | ✅        |
| Connector version semver check                                          | ✅        |
| Connector version increment check                                       | ✅        |
| QA checks                                                               | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/mal_pipedrive-pr-29293/1695247435/d97b030851f257554e82599134b10efebec8b571/source-pipedrive/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d97b030851f257554e82599134b10efebec8b571)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-pipedrive test
```

File: airbyte-integrations/connectors/source-pipedrive/source_pipedrive/manifest.yaml [hunk 4]
```diff
@@ -242,7 +242,7 @@ definitions:
       path_extractor: "product"
       path: "/recents"
 
-  products_fields_stream:
+  product_fields_stream:
     $ref: "#/definitions/base_stream"
     name: "products_fields"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-pipedrive/source_pipedrive/manifest.yaml:247]: You need to change here too.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-pipedrive test report (commit [`e405312fe7`](https://github.com/airbytehq/airbyte/pull/30553/commits/e405312fe7a597ee21cbe2da1af513543880300e)) - ✅

⏲️  Total pipeline duration: 02mn45s 

| Step                                                                    | Result   |
|:------------------------------------------------------------------------|:---------|
| Connector package install                                               | ✅        |
| Build source-pipedrive docker image for platform linux/x86_64           | ✅        |
| Acceptance tests                                                        | ✅        |
| Code format checks                                                      | ✅        |
| Validate airbyte-integrations/connectors/source-pipedrive/metadata.yaml | ✅        |
| Connector version semver check                                          | ✅        |
| Connector version increment check                                       | ✅        |
| QA checks                                                               | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/mal_pipedrive-pr-29293/1695248377/e405312fe7a597ee21cbe2da1af513543880300e/source-pipedrive/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e405312fe7a597ee21cbe2da1af513543880300e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-pipedrive test
```

**Review by marcosmarxm — COMMENTED** [review]
Mal only need to check this [thread](https://airbytehq-team.slack.com/archives/C02U9R3AF37/p1695299055330149) about the field `id` inside the  `product_fields` schema. After solved that LGTM everything is workign great :)

**Review by marcosmarxm — APPROVED** [review]
LGTM

**Pull request merged**